internal/handlers/common/filter.go

```diff
@@ -289,17 +299,31 @@ func filterFieldExpr(fieldValue any, expr *types.Document) (bool, error) {
 
 // filterFieldRegex handles {field: /regex/} filter.
 func filterFieldRegex(fieldValue any, regex types.Regex) (bool, error) {
-	s, ok := fieldValue.(string)
-	if !ok {
-		return false, nil
-	}
-
 	re, err := regex.Compile()
 	if err != nil {
 		return false, err
 	}
 
-	return re.MatchString(s), nil
+	switch fieldValue := fieldValue.(type) {
+	case string:
+		return re.MatchString(fieldValue), nil
+
+	case *types.Array:
```

Comment:
what cases are being processed here? It's probably not only `{field: /regex/}` ?

Reply:
yes, thank you, I mean to add this example to comments above.

Reply:
`// filterFieldRegex handles {field: /regex/} filter. Provides regular expression capabilities for pattern matching strings  in queries, even if the strings are in an array.`
@seeforschauer, is it okay to add such a comment?

Reply:
yes, thank you it makes cases clear